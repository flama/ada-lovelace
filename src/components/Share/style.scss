@import '../../config';

#share {
  width: 100%;
  overflow: auto;
  background-color: rgba(#000, 0.2);

  > .content {
    @include wrap-mobile;
    @include clearfix;

    @include large-up {
      @include grid(8);
      @include offsetLeft(2);
    }

    padding-top: $space-l;
    padding-bottom: $space-l;

    > h3 {
      margin-bottom: $space-xxs;
      font-size: $font-size-kilo;
      line-height: $line-height-kilo;

      @include large-up {
        font-size: $font-size-hecto;
        line-height: $line-height-hecto;
      }
    }

    > p {
      margin-bottom: $space-m;
      font-size: $font-size-base;
      line-height: $line-height-base;
      color: #fff;
      opacity: 0.3;

      @include large-up {
        margin-bottom: $space-s;
        font-size: $font-size-deci;
        line-height: $line-height-deci;
      }
    }

    button, .SocialMediaShareButton {
      display: inline-block;
      text-align: center;
      padding-top: $space-xxs;
      padding-bottom: $space-xxs;

      @include medium-up {
        padding-left: $space-s;
        padding-right: $space-s;
        padding-top: 0;
        padding-bottom: 0;
        margin-left: $space-xs;
      }

      border-style: solid;
      border-width: 1px;
      border-color: $light-purple;
      border-radius: $space-xxxs;

      font-size: $font-size-centi;
      line-height: $line-height-base;
      letter-spacing: 0.44px;
      color: $light-purple;
      background-color: transparent;
      cursor: pointer;

      transition: all 0.3s ease;

      &:first-child { margin-left: 0; }

      @include small-only {
        &:nth-child(1) {
          width: 48%;
          float: left;
        }

        &:nth-child(2) {
          width: 48%;
          float: right;
        }
      }


      &:hover { color: #fff; }

      &.selected {
        border-color: transparent;
        background-color: $selected;
        color: #fff;
      }
    }

  }
}
