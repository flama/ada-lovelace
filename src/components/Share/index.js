import React from 'react'
import { Element } from 'react-scroll'

import './style.scss'

const Share = props => {
  return (
    <Element name="share" id="share">
      <div className="content">
        <h3>Keep Inspiring!</h3>
        <p>
          { `We envision an ever-growing platform, crowdsourced by people everywhere, helping us show women who somehow have created an impact in our world!
          ` }
        </p>
        <p>
          {`Just click bellow and you can tell us who are the amazing women that inspire you and made a dent in our world.`}
        </p>
        <a href="https://flama.typeform.com/to/RHoEAz" target="_blank" className="button">collaborate</a>
      </div>
    </Element>
  )
}

export default Share
