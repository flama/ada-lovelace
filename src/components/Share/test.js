import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import Share from '../Share'

describe("the share section", () => {
  let share

  beforeEach(() => {
    share = shallow(<Share />)
  })

  it("renders correctly", () => {
    expect(toJSON(share)).toMatchSnapshot()
  })
})
