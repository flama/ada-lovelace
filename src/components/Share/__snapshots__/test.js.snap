// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the share section renders correctly 1`] = `
<_
  id="share"
  name="share"
>
  <div
    className="content"
  >
    <h3>
      Share
    </h3>
    <p>
      We envision an ever-growing platform, crowdsourced by people everywhere, helping us show women who somehow have created an impact in our world!
               Until then, you can help us by spreading the word of Wikimina :)
    </p>
    <div
      className="actions"
    >
      <CreatedButton
        description="Inspiring and helping to create representativity so more women become badasses."
        url="https://flama.github.io/ada-lovelace"
        windowHeight={400}
        windowWidth={550}
      >
        Facebook
      </CreatedButton>
      <CreatedButton
        hashtags={
          Array [
            "wikimina",
            "grrlpower",
            "whyhistorymatters",
            "empoweringwomen",
          ]
        }
        title="Wikimina"
        url="https://flama.github.io/ada-lovelace"
        via="studioflama"
        windowHeight={400}
        windowWidth={550}
      >
        Twitter
      </CreatedButton>
    </div>
  </div>
</_>
`;
